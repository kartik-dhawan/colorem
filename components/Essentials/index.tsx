import { Grid, Typography } from "@mui/material"
import { Box } from "@mui/system"
import { Roboto_Condensed } from "next/font/google"
import { useSelector } from "react-redux"
import { RootType } from "../../redux/constants/stateTypes"
import { EssentialItemsType } from "../../utils/interfaces"
import { styles } from "./styles"

const roboto = Roboto_Condensed({
  subsets: ["latin"],
  weight: ["300", "400", "700"],
  style: ["normal", "italic"],
  display: "swap",
})

const Essentials = () => {
  const eid = "essentialsSection"
  const content = useSelector((state: RootType) => state.contentSlice.data)

  return (
    <Box
      className={eid + "Wrapper"}
      id={eid + "Wrapper"}
      data-testid={eid + "Wrapper"}
      sx={styles.essentialsSectionWrapper}
    >
      <Box
        className={eid + "Title " + roboto.className}
        id={eid + "Title"}
        sx={styles.essentialsSectionTitle}
      >
        The Essentials
      </Box>
      <Grid container className={eid + "Body"} id={eid + "Body"}>
        {content.essentialItemsText?.map((item: EssentialItemsType) => {
          return (
            <Grid
              item
              xs={12}
              md={6}
              lg={4}
              className={eid + "BodyItem"}
              id={eid + "BodyItem_" + item.id}
              sx={styles.essentialsSectionBodyItem}
              key={item.id}
            >
              <Typography variant="h2" sx={styles.essentialsItemTitle}>
                {item.id}
              </Typography>
              <Typography variant="body2" sx={styles.essentialsItemBody}>
                {item.body}
              </Typography>
            </Grid>
          )
        })}
      </Grid>
    </Box>
  )
}

export default Essentials
