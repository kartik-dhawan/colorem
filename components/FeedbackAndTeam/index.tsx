import { Grid } from "@mui/material"
import Feedback from "./Feedback"
import TeamDetails from "./TeamDetails"

const FeedbackAndTeam = () => {
  return (
    <Grid container spacing={5} mt={3}>
      <Feedback />
      <TeamDetails />
    </Grid>
  )
}

export default FeedbackAndTeam
