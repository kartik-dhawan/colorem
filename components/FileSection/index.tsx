import { Grid } from "@mui/material"
import FileCard from "../common/FileCard"

const FileSection = () => {
  const fid = "fileSection"

  return (
    <Grid
      container
      className={fid + "Wrapper"}
      direction="row"
      data-testid={fid + "Wrapper"}
    >
      <FileCard type="right" fid={fid} fr={8} />
      <FileCard type="left" fid={fid} fr={4} />
    </Grid>
  )
}

export default FileSection
