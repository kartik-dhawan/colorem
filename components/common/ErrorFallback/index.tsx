import { Alert, AlertTitle, Box, Button } from "@mui/material"
import { useCallback, useState } from "react"
import { logger } from "../../../lib/methods"
import { styles } from "./styles"

export const myErrorHandler = (
  error: Error,
  info: { componentStack: string }
) => {
  // Do something with the error
  // E.g. log to an error logging client here
  logger({ error, type: "Error" })
  logger({ info, type: "Info" })
}

const ErrorFallback = ({
  error,
  resetErrorBoundary,
}: {
  error: Error
  resetErrorBoundary: any // eslint-disable-line
}) => {
  const efid = "errorFallback"

  const [showErrorDetails, setShowErrorDetails] = useState<boolean>(false)

  const errorDetailsHandler = useCallback(() => {
    setShowErrorDetails(!showErrorDetails)
  }, [showErrorDetails])

  return (
    <Box
      role="alert"
      className={efid + "Wrapper"}
      id={efid + "Wrapper"}
      sx={styles.errorFallbackWrapper}
    >
      <Alert
        severity="error"
        className={efid + "PrimaryAlert"}
        id={efid + "PrimaryAlert"}
        action={
          <Button color="inherit" size="small" onClick={errorDetailsHandler}>
            {showErrorDetails ? "Hide details" : "Show details"}
          </Button>
        }
        sx={styles.errorFallbackHeaderAlert}
      >
        <AlertTitle>Code Error</AlertTitle>
        {error.name}
      </Alert>
      {showErrorDetails && (
        <Alert
          severity="error"
          sx={{
            width: "80%",
          }}
          className={efid + "SecondaryAlert"}
          id={efid + "SecondaryAlert"}
        >
          <AlertTitle>{error.name}</AlertTitle>
          <pre>{error.message}</pre>
          <pre>{error.stack}</pre>
        </Alert>
      )}
      <Box sx={styles.errorFallbackTryAgainBtnWrapper}>
        <Button
          className={efid + "TryAgainBtn"}
          id={efid + "TryAgainBtn"}
          sx={styles.errorFallbackTryAgainBtn}
          onClick={resetErrorBoundary}
        >
          Try Again
        </Button>
      </Box>
    </Box>
  )
}

export default ErrorFallback
