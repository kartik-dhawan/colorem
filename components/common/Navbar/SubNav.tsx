import { Button, Grid } from "@mui/material"
import { useState } from "react"
import { ContentfulType } from "../../../utils/interfaces"
import { styles } from "./styles"
import SideNav from "../../SideNav"

const SubNav = ({ content }: ContentfulType) => {
  const snid = "subNav"
  const [sideNavToggle, setSideNavToggle] = useState<boolean>(false)

  return (
    <Grid
      container
      sx={{
        ...styles.nav,
        margin: "25px 0px",
      }}
      className={snid + "Wrapper"}
      id={snid + "Wrapper"}
    >
      {/* positioned absolute-fixed and will cover the whole screen */}
      <SideNav
        sideNavToggle={sideNavToggle}
        setSideNavToggle={setSideNavToggle}
      />
      <Grid
        item
        xs={6}
        sm={4}
        className={snid + "Title"}
        id={snid + "Title"}
        sx={styles.subNavTitle}
      >
        {content.subnavHeader}
      </Grid>
      <Grid
        item
        xs={6}
        sm={4}
        textAlign={"right"}
        sx={{ color: "#ebe1e1" }} // text_third
        className={snid + "ChooseTheme"}
        id={snid + "ChooseTheme"}
      >
        Theme
      </Grid>
      <Grid item xs={0} sm={4} textAlign={"right"}>
        <Button
          variant="contained"
          className={snid + "GetStartBtn"}
          id={snid + "GetStartBtn"}
          sx={styles.subNavButton}
          onClick={() => {
            console.log("Toggled sidenav")
            setSideNavToggle(!sideNavToggle)
          }}
        >
          Get Started
        </Button>
      </Grid>
    </Grid>
  )
}

export default SubNav
