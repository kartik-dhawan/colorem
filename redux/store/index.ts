import { configureStore } from "@reduxjs/toolkit"
import countSlice from "../slices/countSlice"

const store = configureStore({
  reducer: {
    countSlice,
  },
})

export default store
