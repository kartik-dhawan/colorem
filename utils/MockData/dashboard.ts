export const contentfulDefaultData = {
  subnavHeader: "PERMUTATIONS WITH COLORS AND MORE.",
  browseCardContent: "Because your project needs more colors.",
  feedbackTitle: "WORKING ON YOUR FEEDBACKS!",
  commentFormatText: "sent a feedback",
  feedbackFormatText: "sent a comment",
  teamSectionBody:
    "The super fast color palettes generator! Create the perfect color palette or get inspired by thousands of beautiful color schemes. Explore trending palettes now & start creating.",
  essentialItemsText: [
    {
      id: 1,
      body: "Generate a random palette for your web design. Save it if you like it else hop on to the next.",
    },
    {
      id: 2,
      body: "Bored of browsing palettes? Make your own with a combination of your creativity and our dataset.",
    },
    {
      id: 3,
      body: "We have pre-built templates for color gradients; ready for you to use & make your website radiant.",
    },
  ],
  footerReferenceText:
    "This web app is only made as a project and in complete reference to the following resources...",
  footerDevelopedByText: "Coded & developed completely by ",
  developedBy: "Kartik Dhawan",
  references: [
    {
      type: "UI/UX Design",
      title: "Halo Lab",
      source:
        "https://dribbble.com/shots/18365166-Web-site-design-landing-page-home-page-ui",
      sourceName: "Dribbble",
    },
    {
      type: "Web Application",
      title: "Coolers",
      source: "https://coolors.co/",
      sourceName: "Coolers",
    },
    {
      type: "API",
      title: "Colormind",
      source: "http://colormind.io/api-access/",
      sourceName: "Colormind API",
    },
  ],
  teamMembers: [
    {
      metadata: { tags: [] },
      sys: {
        space: {
          sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
        },
        id: "5L8OizpXMHyrYgM5ao6NTx",
        type: "Asset",
        createdAt: "2022-12-08T22:01:54.500Z",
        updatedAt: "2022-12-08T22:01:54.500Z",
        environment: {
          sys: { id: "master", type: "Link", linkType: "Environment" },
        },
        revision: 1,
        locale: "en-US",
      },
      fields: {
        title: "Kartik Dhawan",
        description: "Avatar of Kartik Dhawan",
        file: {
          url: "//images.ctfassets.net/1v98waf66sfj/5L8OizpXMHyrYgM5ao6NTx/013c075eacd65ad632e43442e0b620c6/user1.JPG",
          details: { size: 34091, image: { width: 400, height: 400 } },
          fileName: "user1.JPG",
          contentType: "image/jpeg",
        },
      },
    },
    {
      metadata: { tags: [] },
      sys: {
        space: {
          sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
        },
        id: "3gNCsKWDkm9rWiNkurM5Te",
        type: "Asset",
        createdAt: "2022-12-08T22:02:45.218Z",
        updatedAt: "2022-12-08T22:02:45.218Z",
        environment: {
          sys: { id: "master", type: "Link", linkType: "Environment" },
        },
        revision: 1,
        locale: "en-US",
      },
      fields: {
        title: "Gretchen Morgan",
        description: "Avatar of Gretchen Morgan",
        file: {
          url: "//images.ctfassets.net/1v98waf66sfj/3gNCsKWDkm9rWiNkurM5Te/579d621f150b8f9ebf301e8a27fce0e6/user2.PNG",
          details: { size: 411188, image: { width: 500, height: 500 } },
          fileName: "user2.PNG",
          contentType: "image/png",
        },
      },
    },
    {
      metadata: { tags: [] },
      sys: {
        space: {
          sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
        },
        id: "43okVLERYKi8ibgIVdWDpb",
        type: "Asset",
        createdAt: "2022-12-08T22:03:17.269Z",
        updatedAt: "2022-12-08T22:03:17.269Z",
        environment: {
          sys: { id: "master", type: "Link", linkType: "Environment" },
        },
        revision: 1,
        locale: "en-US",
      },
      fields: {
        title: "Dunder Mifler",
        description: "Avatar of Dunder Mifler",
        file: {
          url: "//images.ctfassets.net/1v98waf66sfj/43okVLERYKi8ibgIVdWDpb/6c4334b2e82b16d038df934d114ee16d/user3.JPG",
          details: { size: 64676, image: { width: 623, height: 615 } },
          fileName: "user3.JPG",
          contentType: "image/jpeg",
        },
      },
    },
  ],
  sideNavigationItems: [
    { id: 1, body: "Go back to landing page.", link: "/", title: "Home" },
    {
      id: 2,
      body: "Browse through the color palettes for web-aesthetic.",
      link: "/palettes",
      title: "Palettes",
    },
    {
      id: 3,
      body: "Browse pre-built templates for color gradients.",
      link: "/gradients",
      title: "UI Gradiants",
    },
    {
      id: 4,
      body: "Go deep into the technical side of the project",
      link: "/about/more",
      title: "More on the project",
    },
  ],
  quickLinks: [
    { id: 1, path: "/about", title: "About" },
    { id: 2, path: "/about/more", title: "More" },
    { id: 3, path: "/palettes/create", title: "Create a palette" },
  ],
  paletteSectionInstructor: [
    {
      id: 1,
      icon: "spacebar",
      instruction:
        "Hit spacebar or next palette icon next to 'Like' button to go to next palette.",
    },
    {
      id: 2,
      icon: "tap",
      instruction:
        "Click/tap on the hex-code written in any individual palette to copy it",
    },
  ],
  gradientSectionInstructor: [
    {
      id: 1,
      icon: "tap",
      instruction: "Hover/Tap on the gradient to view available actions.",
    },
    {
      id: 2,
      icon: "info",
      instruction:
        "Click on 'info' icon to get more information & code of the gradient.",
    },
  ],
  customErrorPagesData: {
    "404": {
      id: 1,
      statusBody: "The page you're looking for doesn't exist.",
      statusCode: "404",
      statusTitle: "Uncharted waters!",
    },
    redirectBtnText: "Get back to dashboard?",
  },
  defaultMetaImage: {
    metadata: { tags: [] },
    sys: {
      space: {
        sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
      },
      id: "3AZSHDIcX2l5wtugFxRmWn",
      type: "Asset",
      createdAt: "2023-01-31T19:35:42.698Z",
      updatedAt: "2023-01-31T20:21:25.407Z",
      environment: {
        sys: { id: "master", type: "Link", linkType: "Environment" },
      },
      revision: 5,
      locale: "en-US",
    },
    fields: {
      title: "Home Meta Image",
      description: "",
      file: {
        url: "//images.ctfassets.net/1v98waf66sfj/3AZSHDIcX2l5wtugFxRmWn/0d56e19f5f34ca6dd7a1ff453cefdcf4/homeMetaImage.webp",
        details: { size: 13532, image: { width: 1090, height: 1090 } },
        fileName: "homeMetaImage.webp",
        contentType: "image/webp",
      },
    },
  },
  aboutPageNavItems: [
    { id: 1, route: "project", title: "About the project" },
    { id: 2, route: "dev", title: "Behind the desk" },
    { id: 3, route: "tech", title: "Tech nuggets" },
    { id: 4, route: "team", title: "The colorem team" },
    { id: 5, route: "more", title: "More" },
  ],
  aboutPageDescription:
    "A little too much information on this page. Make the tone warmer and get you reading glasses to read more about the project.\n",
  portfolioUrl: "https://kartikdhawan.vercel.app",
}
