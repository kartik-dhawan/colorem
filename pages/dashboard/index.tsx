import { Box } from "@mui/material"
import SubNav from "../../components/common/Navbar/SubNav"
import FileSection from "../../components/FileSection"
import Essentials from "../../components/Essentials"
import FeedbackAndTeam from "../../components/FeedbackAndTeam"
import { client } from "../../utils/contentful/config"
import { ContentfulType } from "../../utils/interfaces"
import { useEffect } from "react"
import { useDispatch } from "react-redux"
import { updateContent } from "../../redux/slices/contentSlice"

export const getStaticProps = async () => {
  const response = await client.getEntries({ content_type: "coloremDashboard" }) // eslint-disable-line

  return {
    props: {
      data: response?.items[0]?.fields,
    },
  }
}

const Dashboard = ({ data }: ContentfulType) => {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(updateContent(data))
  }, [data])

  return (
    <Box>
      <SubNav content={data} />
      <FileSection />
      <FeedbackAndTeam />
      <Essentials />
    </Box>
  )
}

export default Dashboard
