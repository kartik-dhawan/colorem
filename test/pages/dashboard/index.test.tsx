import React from "react"
import renderer from "react-test-renderer"
import { ErrorBoundary } from "react-error-boundary"
import { Box } from "@mui/material"
import SubNav from "../../../components/common/Navbar/SubNav"
import Essentials from "../../../components/Essentials"
import FeedbackAndTeam from "../../../components/FeedbackAndTeam"
import FileSection from "../../../components/FileSection"
import ErrorFallback, {
  myErrorHandler,
} from "../../../components/common/ErrorFallback"
import { Provider } from "react-redux"
import store from "../../../redux/store"
import { contentfulDefaultData } from "../../../utils/MockData/dashboard"

describe("Dashboard page", () => {
  it("should render", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Box>
            <ErrorBoundary
              FallbackComponent={ErrorFallback}
              onError={myErrorHandler}
            >
              <SubNav content={contentfulDefaultData} />
              <FileSection />
              <FeedbackAndTeam />
              <Essentials />
            </ErrorBoundary>
          </Box>
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
