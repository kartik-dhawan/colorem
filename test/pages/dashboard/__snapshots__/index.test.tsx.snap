// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard page should render 1`] = `
<div
  className="MuiBox-root css-0"
>
  <div
    className="MuiGrid-root MuiGrid-container subNavWrapper css-1oudet0-MuiGrid-root"
    id="subNavWrapper"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 subNavTitle className css-3wmuhz-MuiGrid-root"
      id="subNavTitle"
    >
      PERMUTATIONS WITH COLORS AND MORE.
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-2 subNavChooseTheme css-9bgm2f-MuiGrid-root"
      id="subNavChooseTheme"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DarkModeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
          />
        </svg>
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 css-4q8h96-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium subNavGetStartBtn css-ubt6ix-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        id="subNavGetStartBtn"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Start looking
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container fileSectionWrapper css-11lq3yg-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 fileSectionCard css-k0o1dg-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item fileSectionCardLower css-1hjlv0p-MuiGrid-root"
        id="fileSectionCardLower"
      >
        <div
          className="fileSectionRightFileContent"
        >
          <div
            className="fileSectionContent"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium fileSectionGrainIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="GrainIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 12c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zM6 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12-8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm-4 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
            <h5
              className="MuiTypography-root MuiTypography-h5 fileSectionContentText css-ag7rrr-MuiTypography-root"
            />
          </div>
          <div
            className="fileSectionRightFileImage"
          >
            <img
              alt="bitmoji image"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height={425}
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimages%2Fbrowse_bitmoji.webp&w=1080&q=75"
              srcSet="/_next/image?url=%2Fimages%2Fbrowse_bitmoji.webp&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fbrowse_bitmoji.webp&w=1080&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={425}
            />
          </div>
          <a
            className="fileSectionArrowIconWrapper"
            href="/gradients"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <div
              className="fileSectionBrowse"
            >
              Browse
            </div>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium fileSectionArrowIcon css-i4bv87-MuiSvgIcon-root"
              data-testid="EastIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 fileSectionCard css-1eqpqzh-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item fileSectionCardLower css-5w37io-MuiGrid-root"
        id="fileSectionCardLower"
      >
        <div
          className="MuiContainer-root MuiContainer-maxWidthLg css-r2ks57-MuiContainer-root"
        >
          <span
            className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-4i0lvv-MuiSkeleton-root"
            style={
              {
                "height": 56,
                "width": "100%",
              }
            }
          />
          <span
            className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-ovvgps-MuiSkeleton-root"
            style={
              {
                "height": 56,
                "width": "100%",
              }
            }
          />
          <span
            className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-aedbwi-MuiSkeleton-root"
            style={
              {
                "height": 56,
                "width": "100%",
              }
            }
          />
          <span
            className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-2ql2a-MuiSkeleton-root"
            style={
              {
                "height": 56,
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1n8pt8h-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 feedbackSectionWrapper css-14expfb-MuiGrid-root"
      id="feedbackSectionWrapper"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 feedbackSectionTitle css-1pcczsu-MuiTypography-root"
        id="feedbackSectionTitle"
      />
      <div
        className="feedbackCardsWrapper MuiBox-root css-0"
        id="feedbackCardsWrapper"
      >
        <div
          className="feedbackCard MuiBox-root css-hkx9zj"
          id="feedbackCard"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium feedbackIcon css-i4bv87-MuiSvgIcon-root"
            data-testid="FavoriteIcon"
            focusable="false"
            id="feedbackIcon"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1 css-rn997f-MuiTypography-root"
          >
            23
             new likes on palettes.
          </p>
        </div>
        <div
          className="feedbackCard MuiBox-root css-hkx9zj"
          id="feedbackCard"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium feedbackIcon css-i4bv87-MuiSvgIcon-root"
            data-testid="PersonOutlineIcon"
            focusable="false"
            id="feedbackIcon"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1 feedbackCardBody css-rn997f-MuiTypography-root"
          >
            Mr. Shay sent a
             
            <b>
              <i>
                feedback
              </i>
            </b>
            , 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores eaque dolor ad! ðŸ‡¦ðŸ‡· ðŸ’Ÿ'
          </p>
        </div>
        <div
          className="feedbackCard MuiBox-root css-hkx9zj"
          id="feedbackCard"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium feedbackIcon css-i4bv87-MuiSvgIcon-root"
            data-testid="CommentIcon"
            focusable="false"
            id="feedbackIcon"
            viewBox="0 0 24 24"
          >
            <path
              d="M21.99 4c0-1.1-.89-2-1.99-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4-.01-18zM18 14H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
            />
          </svg>
          <p
            className="MuiTypography-root MuiTypography-body1 feedbackCardBody css-rn997f-MuiTypography-root"
          >
            Mr. Shay sent a
             
            <b>
              <i>
                comment
              </i>
            </b>
            , 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores eaque dolor ad! ðŸ‡¦ðŸ‡· ðŸ’Ÿ'
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 teamSectionWrapper css-16629bz-MuiGrid-root"
      id="teamSectionWrapper"
    >
      <div
        className="teamSectionHead MuiBox-root css-w09u1u"
        id="teamSectionHead"
      >
        <a
          className="teamSectionTitle"
          href="/about/team"
          id="teamSectionTitle"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-fv8tu3-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            TEAM
          </button>
        </a>
        <div
          className="teamSectionAvatarWrapper MuiBox-root css-1ra01ct"
          id="teamSectionAvatarWrapper"
        />
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 teamSectionBody css-1tz1ww6-MuiTypography-root"
        id="teamSectionBody"
      />
    </div>
  </div>
  <div
    className="essentialsSectionWrapper MuiBox-root css-1q5wgsi"
    id="essentialsSectionWrapper"
  >
    <div
      className="essentialsSectionTitle className MuiBox-root css-ns0x91"
      id="essentialsSectionTitle"
    >
      The Essentials
    </div>
    <div
      className="MuiGrid-root MuiGrid-container essentialsSectionBody css-11lq3yg-MuiGrid-root"
      id="essentialsSectionBody"
    />
  </div>
</div>
`;
